spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/JdbcTypeFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jdbc.core.convert;
+
+import java.sql.Array;
+
+/**
+ * Allows the creation of instances of database dependent types, e.g. {@link Array}.
+ *
+ * @author Jens Schauder
+ * @since 1.1
+ */
+public interface JdbcTypeFactory {
+
+	/**
+	 * A dummy implementation used in places where a proper {@code JdbcTypeFactory} can not be provided but an instance
+	 * needs to be provided anyway, mostly for providing backward compatibility. Calling it will result in an exception.
+	 * The features normally supported by a {@link JdbcTypeFactory} will not work.
+	 */
+	JdbcTypeFactory DUMMY_JDBC_TYPE_FACTORY = value -> {
```

Comment:
How about naming this one `SIMPLE`? From a design perspective, we could turn the field into a static interface method to align with our API style.

Reply:
I don't like `SIMPLE` because it is kind of less than simple: Not really working. How about `UNSUPPORTED`?